framework/src/Volo.Abp.BackgroundJobs.HangFire/Volo/Abp/BackgroundJobs/Hangfire/HangfireJobExecutionAdapter.cs

```diff
@@ -1,27 +1,38 @@
 ﻿using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
 using Microsoft.Extensions.Options;
 using Volo.Abp.Threading;
 
 namespace Volo.Abp.BackgroundJobs.Hangfire
 {
     public class HangfireJobExecutionAdapter<TArgs>
     {
+        public ILogger<HangfireJobExecutionAdapter<TArgs>> Logger { get; set; }
+        
         protected AbpBackgroundJobOptions Options { get; }
         protected IServiceScopeFactory ServiceScopeFactory { get; }
         protected IBackgroundJobExecuter JobExecuter { get; }
 
         public HangfireJobExecutionAdapter(
-            IOptions<AbpBackgroundJobOptions> options, 
-            IBackgroundJobExecuter jobExecuter, 
+            IOptions<AbpBackgroundJobOptions> options,
+            IBackgroundJobExecuter jobExecuter,
             IServiceScopeFactory serviceScopeFactory)
         {
             JobExecuter = jobExecuter;
             ServiceScopeFactory = serviceScopeFactory;
             Options = options.Value;
+            Logger = NullLogger<HangfireJobExecutionAdapter<TArgs>>.Instance;
         }
 
         public void Execute(TArgs args)
         {
+            if (!Options.IsJobExecutionEnabled)
+            {
+                Logger.LogWarning("Background jobs system is disabled");
+                return;
```

Comment:
When you return, what happens to the job data? Is it lost? I suppose so. Is so, this is dangerous. A better implementation should completely disable hangfire for this application instance, so it does not call the Execute method at all and the job data waits in the database for another application to execute it. What do you think?

Reply:
I agree that there may be a risk of losing job data. 
But we should not completely disable hangfire, because for abp `Volo.Abp.Hangfire` **!=** `Volo.Abp.BackgourndJob.Hangfire`, hangfire may be used as a background worker integration. When we disable the background job system, we don't want to disable the background worker system. I will try to find other solutions.

Reply:
I believe that we can disable backgroung jobs & workers together, no problem, because they are related to each other and we normally want to disable together.

Reply:
If it is disabled together, then we don't need two enabled switches: `IsEnabled` and `IsJobExecutionEnabled`. Just one switch is enough